pkg/services/oauthtoken/oauth_token.go

```diff
@@ -34,11 +34,11 @@ var (
 const maxOAuthTokenCacheTTL = 10 * time.Minute
 
 type Service struct {
-	Cfg               *setting.Cfg
-	SocialService     social.Service
-	AuthInfoService   login.AuthInfoService
-	singleFlightGroup *singleflight.Group
-	cache             *localcache.CacheService
+	Cfg             *setting.Cfg
+	SocialService   social.Service
+	AuthInfoService login.AuthInfoService
+	serverLock      *serverlock.ServerLockService
+	cache           remotecache.CacheStorage
```

Comment:
It appears as though since we'll be using the server lock for preventing simultaneous refreshes, we don't gain much by switching to a remote cache, since the latter is used for skipping unnecessary refreshes if a token hasn't been expired yet.

My (superficial) understanding is that this hook gets called on every request and switching to the remote cache could potentially result unnecessary load on the database (if the remote cache is configured to use the db and not redis/memcached).

Reply:
Good call-out, there certainly nothing stopping us from reverting the cache changes if that's the case.

Reply:
After reading the code I'm actually struggling to understand what the cache actually does for us. `buildOAuthTokenFromAuthInfo` is always called regardless of whether the cache has been populated, and checking whether the token needs to be refreshed is also relatively cheap. I may be missing something here, but it seems like this code could be simplified quite a bit and made easier to reason about if we just remove the cache altogether.